src/lib/dnssd/Discovery_ImplPlatform.cpp

```diff
@@ -240,7 +240,8 @@ CHIP_ERROR CopyTxtRecord(TxtFieldKey key, char * buffer, size_t bufferLen, const
     switch (key)
     {
     case TxtFieldKey::kTcpSupported:
-        return CopyTextRecordValue(buffer, bufferLen, params.GetTcpSupported());
+        VerifyOrReturnError(params.GetTCPSupportModes() != TCPModeAdvertise::kNone, CHIP_ERROR_UNINITIALIZED);
+        return CopyTextRecordValue(buffer, bufferLen, static_cast<uint16_t>(params.GetTCPSupportModes()));
```

Comment:
why uint16_t? why not to_underlying usage?

Reply:
The CopyTextRecordValue() is an overloaded function but it did not have one for uint8_t. Converted the enum to uint16_t and changed it to to_underlying() over here.